internal/store/memory/store.go

```diff
@@ -222,7 +198,7 @@ func (s *Store) UpsertRoute(ctx context.Context, route store.Route, updateCondit
 
 	// bind to gateways
 	for _, gateway := range s.gateways {
-		gateway.TryBind(ctx, route)
+		gateway.Bind(ctx, route)
 	}
 
 	// sync the gateways to consul and route statuses to k8s
```

Comment:
Just for posterity sake, the optimization of only syncing based off of ingress gateway changes comes from trying to ensure the `Sync` call below this doesn't hammer Consul every time any resource is reconciled. In the last part of the refactor we do some smarter dirty checking to make sure we're only updating gateways whose state values have actually changed from their last stored state, so even if we temporarily un-gate the syncs an over-sync to Consul right now, it should be fixed soon enough.

@nathancoleman and I talked about that offline.